src/thor/costmatrix.cc

```diff
@@ -682,15 +682,11 @@ void CostMatrix::SetSources(GraphReader& graphreader,
   uint32_t index = 0;
   Cost empty_cost;
   for (const auto& origin : sources) {
-    // Set up lambda to get sort costs
-    const auto edgecost = [this, index](const uint32_t label) -> float {
-      return source_edgelabel_[index][label].sortcost();
-    };
-
     // Allocate the adjacency list and hierarchy limits for this source.
     // Use the cost threshold to size the adjacency list.
-    source_adjacency_[index].reset(
-        new DoubleBucketQueue(0, current_cost_threshold_, costing_->UnitSize(), edgecost));
+    source_adjacency_[index].reset(new DoubleBucketQueue<BDEdgeLabel>(0, current_cost_threshold_,
```

Comment:
we no longer pass a lambda/function to the queue, isntead we give it access to the edge labels directly. one day we should merge the queue the labels and the status objects into one class called queue and do it all from there